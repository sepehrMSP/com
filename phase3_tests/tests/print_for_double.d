void print(double f){
    int cnt;
    f=f+0.00005;
    cnt=0;
    while(f>=10.0){
        f=f/10.0;
        cnt=cnt+1;
    }
    while(cnt>=0){
        int ragham;
        ragham=0;
        while(f>=1.0){
            ragham=ragham+1;
            f=f-1.0;
        }
        Print(ragham);
        f=f*10.0;
        cnt=cnt-1;
    }
    Print(".");
    for(cnt=0;cnt<4;cnt=cnt+1){
        int ragham;
        ragham=0;
        while(f>=1.0){
            ragham=ragham+1;
            f=f-1.0;
        }
        Print(ragham);
        f=f*10.0;
    }

}

int main() {
    print(123456789.12346);
    Print(123456789.12346);
}
