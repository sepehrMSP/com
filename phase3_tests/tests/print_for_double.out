1
2
3
4
5
6
7
8
9
.
1
2
3
5
123456789.1235
